packages/design-system/src/components/Badge/primitive/Badge.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { forwardRef, HTMLAttributes, Ref } from 'react';
+import classNames from 'classnames';
+
+import { ButtonTertiary } from '../../Button';
+import { ButtonIcon } from '../../ButtonIcon';
+import Divider from '../../Divider';
+import Dropdown from '../../Dropdown';
+import { StackHorizontal } from '../../Stack';
+
+import style from './Badge.module.scss';
+
+type BaseBadgeProps = Omit<
+	HTMLAttributes<HTMLDivElement>,
+	'className' | 'defaultValue' | 'onChange' | 'style'
+> & {
+	label: string;
+};
+
+type RemovableBadgeProps = BaseBadgeProps & {
+	onRemove: (event: React.MouseEvent<HTMLButtonElement> | KeyboardEvent) => void;
+};
+
+type BadgeWithDropdownProps = BaseBadgeProps & {
+	values: string[];
+	defaultValue?: string;
+	onChange: (
+		event: React.MouseEvent<HTMLButtonElement> | KeyboardEvent,
+		value: { label: string },
+	) => void;
+};
+
+export type BadgeProps = RemovableBadgeProps | BadgeWithDropdownProps;
```

Comment:
Why can't BadgeWithDropdown be removed?

Reply:
Because they have a limited list of options you must select. i.e. Sort order (asc vs desc)